dashboard/app/models/pd/workshop.rb

```diff
@@ -213,8 +213,30 @@ def self.start_on_or_after(date)
 
   # Orders by the scheduled start date (date of the first session),
   # @param :desc [Boolean] optional - when true, sort descending
-  def self.order_by_start(desc: false)
-    joins(:sessions).group(:pd_workshop_id).order('DATE(MIN(start))' + (desc ? ' DESC' : ''))
+  def self.order_by_scheduled_start(desc: false)
+    joins(:sessions).
+    group('pd_workshops.id').
+    order('DATE(MIN(pd_sessions.start))' + (desc ? ' DESC' : ''))
+  end
+
+  # Orders by the number of active enrollments
+  # @param :desc [Boolean] optional - when true, sort descending
+  def self.order_by_enrollment_count(desc: false)
+    left_outer_joins(:enrollments).
+    group('pd_workshops.id').
+    order('COUNT(pd_enrollments.id)' + (desc ? ' DESC' : ''))
+  end
+
+  # Orders by the workshop state, in order: Not Started, In Progress, Ended
+  # @param :desc [Boolean] optional - when true, sort descending
+  def self.order_by_state(desc: false)
```

Comment:
I also don't feel great that this logic is duplicated in SQL and in the [model](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/workshop.rb#L267-L271), but I couldn't come up with a better solution short of saving the state in the DB and I didn't like that either. This logic is pretty small and unlikely to change so I think it's ok.